tests/common/plugins/dut_monitor/pytest_dut_monitor.py

```
@@ -16,29 +16,6 @@
 DUT_CPU_LOG = "/tmp/cpu.log"
 DUT_RAM_LOG = "/tmp/ram.log"
 DUT_HDD_LOG = "/tmp/hdd.log"
-THRESHOLDS = os.path.join(os.path.split(__file__)[0], "thresholds.yml")
```

Why do you moved this code to the __init__.py file? This code will not be execute in such case.

---

Indeed I refactored this plugin a little bit. In the refactored code, we do not need this global variable THRESHOLDS. The DUTMonitorPlugin is enhanced with a threshold attribute. And in the __init__.py::pytest_configure, I add logic to use the default threshold file.

---

In sonic-mgmt/tests/conftest.py, if we require the plugin like this, the code in __init__.py will be executed:
```
pytest_plugins = ('common.plugins.ptfadapter',
                  'common.plugins.ansible_fixtures',
                  'common.plugins.dut_monitor',
                  'common.plugins.fib',
                  'common.plugins.psu_controller',
                  'common.plugins.loganalyzer')
```

---

Regarding "THRESHOLDS" its ok, here I also mentioned that fixtures code may not be executed in init file.

---

Just make sure that fixture code is executed, you my not see any exception during tests run because its autouse fixtures, and it can just be silently skipped(not executed).

---

Yes, the fixture code can be executed, I have verified it. Under the hood, pytest imported the required plugins. When the plugin is imported, the `__init__.py` file is executed and the fixture got defined.